packages/ui/src/components/Dialogs/EditFeesDialog.tsx

```diff
@@ -111,6 +174,393 @@ export default function FeeRecipientDialog({
     return areAllFeeRecipientsEditable(selectedTags);
   }
 
+  function areAllOldFrsSameAsGiven(givenFr: string) {
+    const oldFeeRecipients = selectedRows
+      .map((rowId) => rows[parseInt(rowId.toString())].feeRecipient)
+      .flat();
+
+    console.log(oldFeeRecipients);
+    return oldFeeRecipients.every((fr) => fr === givenFr);
+  }
+  function isRemovingMevSpFr() {
+    const oldFeeRecipients = selectedRows
+      .map((rowId) => rows[parseInt(rowId.toString())].feeRecipient)
+      .flat();
+
+    return (
+      oldFeeRecipients.includes(mevSpAddress) &&
+      isNewFeeRecipientValid() &&
+      newFeeRecipient !== mevSpAddress
+    );
+  }
+
+  function isNewFeeRecipientValid() {
+    return (
+      isValidEcdsaPubkey(newFeeRecipient) && newFeeRecipient !== BURN_ADDRESS
+    );
+  }
+
+  const isAnyFormatValidatorSelected = ({
+    givenFormat,
+    checkEquality,
+  }: {
+    givenFormat: WithdrawalCredentialsFormat;
+    checkEquality: boolean;
+  }): boolean => {
+    let isAnyGiven = false;
+    for (const row of selectedRows) {
+      const withdrawalFormat =
+        rows[parseInt(row.toString())].withdrawalCredentials.format;
+      if (
+        checkEquality
+          ? withdrawalFormat === givenFormat
+          : withdrawalFormat !== givenFormat
+      ) {
+        isAnyGiven = true;
+        break;
+      }
+    }
+    return isAnyGiven;
+  };
+
+  const getSmoothValidatorsSelected = () => {
+    const auxList: string[] = [];
+    selectedRows.forEach((row) => {
+      rows[parseInt(row.toString())].feeRecipient === mevSpAddress &&
+        auxList.push(rows[parseInt(row.toString())].pubkey);
+    });
+    setSmoothValidatorsPubkeys(auxList);
+  };
+
+  const nonEcdsaValidatorsData = () => {
```

Comment:
Same as above, you can try:

```
const nonEcdsaValidatorsData = (): void => {
  const filteredValidators = selectedRows
    .map(row => {
      const rowData = rows[+row];
      return rowData.withdrawalCredentials.format !== "ecdsa" 
        ? { pubkey: rowData.pubkey, withdrawalFormat: rowData.withdrawalCredentials.format }
        : null;
    })
    .filter(item => item !== null);

  setNonEcdsaValidatorsData(filteredValidators);
};

```

You can make any changes, it is just a suggestion